### ISSUE TITLE
Still maintained?

### ISSUE THREAD

**Issue by danielnbalasoiu** [SamuelHassine/zotero-prime#22]
Is this project still maintained. I checked docker-compose.yml file and the entire software suite is using old version (tags).

**Comment by hoehnp** [comment]
@danielnbalasoiu, from my experience I know setting up successfully an own instance of zotero is pretty challenging. Additionally, the focus of the zotero developers is the cloud version and, therefore, any solution like zotero-prime is most likely a community contributed 3rd party niche offer without any funding behind. However, keeping up with the latest versions needs efforts which aren't valuable for anyone who already has a running version. 

TL;DR: I wouldn't expect much updates here. Except community contributed fixes in forks of this repository or in the issues of this repository.

**Comment by hpxing** [comment]
zotero6 is a  huge uptade~~ pls consider upate to 6 thanks

**Comment by Huaweidev** [comment]
> zotero6 is a huge uptade~~ pls consider upate to 6 thanks

Is there any other third party zotero platform which can run on my own server?

**Comment by Huaweidev** [comment]
> @danielnbalasoiu, from my experience I know setting up successfully an own instance of zotero is pretty challenging. Additionally, the focus of the zotero developers is the cloud version and, therefore, any solution like zotero-prime is most likely a community contributed 3rd party niche offer without any funding behind. However, keeping up with the latest versions needs efforts which aren't valuable for anyone who already has a running version.
> 
> TL;DR: I wouldn't expect much updates here. Except community contributed fixes in forks of this repository or in the issues of this repository.


Why so many fatal error when cloning?
```
fatal: clone of 'git://github.com/zotero/zotero-libreoffice-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-libreoffice-integration' failed
Failed to clone 'modules/zotero-libreoffice-integration'. Retry scheduled
Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration'...
fatal: unable to connect to github.com:
github.com[0: 20.205.243.166]: errno=Connection timed out

fatal: clone of 'git://github.com/zotero/zotero-word-for-mac-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration' failed
Failed to clone 'modules/zotero-word-for-mac-integration'. Retry scheduled
Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-windows-integration'...
```

Finally failed with these errors:
```
Downloading https://files.pythonhosted.org/packages/3a/dc/bf2b15d1fa15a6f7a9e77a61b74ecbbae7258558fcda8ffc9a6638a6b327/docutils-0.15.2-py2-none-any.whl (548kB)
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 353, in run
    wb.build(autobuilding=True)
  File "/usr/lib/python2.7/dist-packages/pip/wheel.py", line 749, in build
    self.requirement_set.prepare_files(self.finder)
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 620, in _prepare_file
    session=self.session, hashes=hashes)
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 821, in unpack_url
    hashes=hashes
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 659, in unpack_http_url
    hashes)
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 902, in _download_http_url
    _download_url(resp, link, content_file, hashes)
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 603, in _download_url
    hashes.check_against_chunks(downloaded_chunks)
  File "/usr/lib/python2.7/dist-packages/pip/utils/hashes.py", line 46, in check_against_chunks
    for chunk in chunks:
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 571, in written_chunks
    for chunk in chunks:
  File "/usr/lib/python2.7/dist-packages/pip/utils/ui.py", line 139, in iter
    for x in it:
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 560, in resp_read
    decode_content=False):
  File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 436, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 401, in read
    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
  File "/usr/lib/python2.7/contextlib.py", line 35, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 316, in _error_catcher
    raise ReadTimeoutError(self._pool, None, 'Read timed out.')
ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
The command '/bin/sh -c DEBIAN_FRONTEND=noninteractive pip install awscli' returned a non-zero code: 2
ERROR: Service 'app-zotero' failed to build : Build failed
```

Any one build it successfully?

**Comment by hoehnp** [comment]
> > @danielnbalasoiu, from my experience I know setting up successfully an own instance of zotero is pretty challenging. Additionally, the focus of the zotero developers is the cloud version and, therefore, any solution like zotero-prime is most likely a community contributed 3rd party niche offer without any funding behind. However, keeping up with the latest versions needs efforts which aren't valuable for anyone who already has a running version.
> > TL;DR: I wouldn't expect much updates here. Except community contributed fixes in forks of this repository or in the issues of this repository.
> 
> Why so many fatal error when cloning?
> 
> ```
> fatal: clone of 'git://github.com/zotero/zotero-libreoffice-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-libreoffice-integration' failed
> Failed to clone 'modules/zotero-libreoffice-integration'. Retry scheduled
> Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration'...
> fatal: unable to connect to github.com:
> github.com[0: 20.205.243.166]: errno=Connection timed out
> 
> fatal: clone of 'git://github.com/zotero/zotero-word-for-mac-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration' failed
> Failed to clone 'modules/zotero-word-for-mac-integration'. Retry scheduled
> Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-windows-integration'...
> ```
> 
> Finally failed with these errors:
> 
> ```
> Downloading https://files.pythonhosted.org/packages/3a/dc/bf2b15d1fa15a6f7a9e77a61b74ecbbae7258558fcda8ffc9a6638a6b327/docutils-0.15.2-py2-none-any.whl (548kB)
> Exception:
> Traceback (most recent call last):
>   File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
>     status = self.run(options, args)
>   File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 353, in run
>     wb.build(autobuilding=True)
>   File "/usr/lib/python2.7/dist-packages/pip/wheel.py", line 749, in build
>     self.requirement_set.prepare_files(self.finder)
>   File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 380, in prepare_files
>     ignore_dependencies=self.ignore_dependencies))
>   File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 620, in _prepare_file
>     session=self.session, hashes=hashes)
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 821, in unpack_url
>     hashes=hashes
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 659, in unpack_http_url
>     hashes)
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 902, in _download_http_url
>     _download_url(resp, link, content_file, hashes)
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 603, in _download_url
>     hashes.check_against_chunks(downloaded_chunks)
>   File "/usr/lib/python2.7/dist-packages/pip/utils/hashes.py", line 46, in check_against_chunks
>     for chunk in chunks:
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 571, in written_chunks
>     for chunk in chunks:
>   File "/usr/lib/python2.7/dist-packages/pip/utils/ui.py", line 139, in iter
>     for x in it:
>   File "/usr/lib/python2.7/dist-packages/pip/download.py", line 560, in resp_read
>     decode_content=False):
>   File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 436, in stream
>     data = self.read(amt=amt, decode_content=decode_content)
>   File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 401, in read
>     raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
>   File "/usr/lib/python2.7/contextlib.py", line 35, in __exit__
>     self.gen.throw(type, value, traceback)
>   File "/usr/share/python-wheels/urllib3-1.22-py2.py3-none-any.whl/urllib3/response.py", line 316, in _error_catcher
>     raise ReadTimeoutError(self._pool, None, 'Read timed out.')
> ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
> The command '/bin/sh -c DEBIAN_FRONTEND=noninteractive pip install awscli' returned a non-zero code: 2
> ERROR: Service 'app-zotero' failed to build : Build failed
> ```
 
> Any one build it successfully?


@Huaweidev: I sometimes rebuild the client and also the last time it worked. Is it really while checking out or what are the exact steps to reproduce?

**Comment by hoehnp** [comment]
> > zotero6 is a huge uptade~~ pls consider upate to 6 thanks
> 
> Is there any other third party zotero platform which can run on my own server?

as I mentioned earlier, most people use the hosted version by the developers. Certainly, it would be great to have an on premises solution. However, maintaining this will probably not be possible with a solid funding, since so far there seems no real interest from the original developers and it is pretty uncertain if this could be viable business model. I would assume the people running it on premises also do it with the intent of cost savings. I know this situation is an issue common to many FOSS projects unfortunately.

**Comment by Huaweidev** [comment]
@hoehnp I do as installation steps, then report errors：
```
$ mkdir /path/to/your/app && cd /path/to/your/app
$ git clone --recursive https://github.com/SamuelHassine/zotero-prime.git
$ cd zotero-prime
$ cd docker
$ sudo docker-compose up -d
```
Is there any other **fully server side B/S mode** literatures manage&read&mark software?

**Comment by hoehnp** [comment]
@Huaweidev, so we talk about the server side code. Then I was mistaken, I was talking about the client side. Sorry. To my knowledge the other large ones are all proprietary, i.e. Mendeley, Citavi, EndNote. Other solutions like bibsonomy, don't have a dedicated client to my knowledge.

**Comment by uniuuu** [comment]
> > @danielnbalasoiu, from my experience I know setting up successfully an own instance of zotero is pretty challenging. Additionally, the focus of the zotero developers is the cloud version and, therefore, any solution like zotero-prime is most likely a community contributed 3rd party niche offer without any funding behind. However, keeping up with the latest versions needs efforts which aren't valuable for anyone who already has a running version.
> > TL;DR: I wouldn't expect much updates here. Except community contributed fixes in forks of this repository or in the issues of this repository.
> 
> Why so many fatal error when cloning?
> 
> ```
> fatal: clone of 'git://github.com/zotero/zotero-libreoffice-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-libreoffice-integration' failed
> Failed to clone 'modules/zotero-libreoffice-integration'. Retry scheduled
> Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration'...
> fatal: unable to connect to github.com:
> github.com[0: 20.205.243.166]: errno=Connection timed out
> 


Hi @Huaweidev 
Because you have got network issue as per your log:  `github.com[0: 20.205.243.166]: errno=Connection timed out`
Try to troubleshoot your network why it cannot connect to the host?

**Comment by uniuuu** [comment]
> > fatal: clone of 'git://github.com/zotero/zotero-libreoffice-integration.git' into submodule path '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-libreoffice-integration' failed
> > Failed to clone 'modules/zotero-libreoffice-integration'. Retry scheduled
> > Cloning into '/home/user/Downloads/zotprime/zotero-prime/client/zotero-standalone-build/modules/zotero-word-for-mac-integration'...
> > fatal: unable to connect to github.com:
> > github.com[0: 20.205.243.166]: errno=Connection timed out
> > ```
> 
> Hi @Huaweidev Because you have got network issue as per your log: `github.com[0: 20.205.243.166]: errno=Connection timed out` Try to troubleshoot your network why it cannot connect to the host?

@Huaweidev 
The issue is not in connection but it's typo in next files  
`./.git/modules/client/zotero-client/config`
`./.git/modules/client/zotero-standalone-build/config`
Instead of `https://` it has `git://` in links which won't work.
Correcting these typos fixes submodule's cloning issue.

**Comment by uniuuu** [comment]
Hi @danielnbalasoiu @hpxing @hoehnp @Huaweidev 
I'm working already on the updating to v.6 and spent first 12 hours. 
You can buy me a coffee, I'll appreciate. https://www.buymeacoffee.com/uniuuu

**Comment by uniuuu** [comment]
Hi folks,

There is pre-alpha version [zotprime v2.0.1](https://github.com/uniuuu/zotprime/tree/v2.0.1)
```
- Client updated to 6.0.26.SOURCE.fb61ea4ba
- Fixed dataserver

```
Testing and comments are welcomed.

**Comment by uniuuu** [comment]
Alpha version has been released: [alpha](https://github.com/uniuuu/zotprime/tree/v2.0.6-alpha)

**Comment by uniuuu** [comment]
Hi folks,

First ZotPrime V2 beta has been released: [ZotPrime v2.0.8-beta](https://github.com/uniuuu/zotprime/tree/v2.0.8-beta)

```
-  Dataserver has been updated to the latest version/commit.
-  Client has been updated to the latest 6.0.27 (CI checks if any new update above 6.0.27)
-  Fixed dataserver issues.
-  Fixed and updated dependencies.
-  Reorganized folder structure.
-  Split tinymceclean-server and stream-server into separate containers.
-  Converted dataserver image to alpine
-  Added automatic client building via docker 
```

Could you please test it and let me know how smooth it's installing? Thank you in advance.

Also please check you may find convenient the proposed way to build client in one command via docker.
`$ DOCKER_BUILDKIT=1 docker build --file client.Dockerfile --output build .`
Built saved into `./build/staging/Zotero_VERSION/`.  Note, currently it builds for Linux OS only.

**Comment by danielnbalasoiu** [comment]
@uniuuu I'm running the stack with docker-compose on a self-hosted VM but didn't got the chance to test it's functionality using the Zotero client since I hadn't a Linux machine at hand. I'll try later today or tomorrow and I'll get back with some feedback.

Until then keep up the great work you're doing 🚀

**Comment by uniuuu** [comment]
> @uniuuu I'm running the stack with docker-compose on a self-hosted VM but didn't got the chance to test it's functionality using the Zotero client since I hadn't a Linux machine at hand. I'll try later today or tomorrow and I'll get back with some feedback.
> 
> Until then keep up the great work you're doing rocket

Hi @danielnbalasoiu 
I have added a feature into script for the case when server is on VM and it requires to indicate server ip instead of localhost refer to https://github.com/FiligranHQ/zotprime/issues/3#issuecomment-499412389.  Now the script is interactive and default value is localhost.

In previous release some PHP bugs were found and a new beta release issued.
Please use this one [v2.5.0-beta)](https://github.com/uniuuu/zotprime/tree/v2.5.0-beta)

```
 ZotPrime 2.5.0-beta
    Dataserver has been updated to the latest version/commit.
    Client has been updated to 6.0.27 (CI checks if any new update above 6.0.27).
    Fixed dataserver issues.
    Fixed and updated dependencies.
    Reorganized folder structure.
    Split tinymceclean-server and stream-server into separate containers.
    Converted dataserver image to alpine.
    Added automatic client building via docker.
    Updated minio to the latest. Latest version has separate Web UI port. Minor change. Minio Web UI is served at port 9001.
    Added possibility to interactively enter a server ip/hostname for a setup with server on VM.
    Dataserver is being copied during image building and won't mount as a volume in docker compose
    Added DEV and PROD docker compose types.
```

**Comment by uniuuu** [comment]
Hi folks, 

The release candidate has been issued [ZotPrime v2.6.1-rc](https://github.com/uniuuu/zotprime/tree/v2.6.1-rc)
For the installation routine please follow the updated README.md.

```
    Dataserver has been updated to the latest version/commit.
    Client has been updated to 6.0.27 (CI checks if any new update above 6.0.27).
    Fixed dataserver issues.
    Fixed and updated dependencies.
    Reorganized folder structure.
    Split tinymceclean-server and stream-server into separate containers.
    Converted dataserver image to alpine.
    Added automatic client building via docker.
    Updated minio to the latest. Latest version has separate Web UI port. Minor change. Minio Web UI is served at port 9001.
    Added possibility to interactively enter a server ip/hostname for a setup with server on VM.
    Dataserver is being copied during image building and won't mount as a volume in docker compose.
    Added DEV and PROD docker compose types.
    Added a single command way to build a client.
    Removed rinetd.
    Added Kubernetes installation in GKE and Microk8s
```
